src/osd/osd_types.h

```diff
@@ -3392,7 +3397,54 @@ WRITE_CLASS_ENCODER(pg_log_entry_t)
 
 ostream& operator<<(ostream& out, const pg_log_entry_t& e);
 
+struct pg_log_dup_t {
+  osd_reqid_t reqid;  // caller+tid to uniquely identify request
+  eversion_t version;
+  version_t user_version; // the user version for this entry
+  int32_t return_code; // only stored for ERRORs for dup detection
+  std::shared_ptr<hobject_t> soid_ref; // since hobject_t's are big, share them
```

Comment:
I'm not so sure about storing these. I know you only did it because of the failing cache-tier tests, but it really inflates the pg_log_dup_t size (I suspect there aren't going to be many repeats in most cache-tier cases?).

Given the use cases we're targeting, I don't think it's reasonable to expect to run base or cache pools without a real pg log. How about we enforce that via reasonable minimums, instead of contorting our cool new memory-slimming data structures to support them? :)